store/LocalStore.js

```diff
@@ -3,37 +3,28 @@
 const lru = require('tiny-lru')
 
 function LocalStore (timeWindow, cache, app, continueExceeding) {
-  this.lru = lru(cache || 5000)
-  this.interval = setInterval(beat.bind(this), timeWindow).unref()
+  this.lru = lru(cache || 5000, timeWindow)
   this.app = app
   this.timeWindow = timeWindow
   this.continueExceeding = continueExceeding
-
-  app.addHook('onClose', (instance, done) => {
-    clearInterval(this.interval)
-    done()
-  })
-
-  function beat () {
-    this.lru.clear()
-    this.msLastBeat = null
-  }
 }
 
 LocalStore.prototype.incr = function (ip, cb) {
-  let current = this.lru.get(ip) || 0
-  this.lru.set(ip, ++current)
+  const nowInMs = Date.now()
```

Comment:
Done this because some tests failed as `x-ratelimit-reset` was not exactly the same second (as it is later rounded):
https://github.com/fastify/fastify-rate-limit/blob/4ad50c6915176db0d7a9ce4cae3f5a2f5a87d82b/test/route-rate-limit.test.js#L46